includes/panopto_derivatives.inc

```diff
@@ -5,37 +5,356 @@
+          $messages = array(
+          'message' => t('Panopto progress error: @exception'),
+          'message_sub' => array('@exception' => $ex->getMessage()),
+          'type' => 'watchdog',
+          'severity' => WATCHDOG_ERROR,
+          );
+          $failureMessage['messages'][] = $messages;
+          watchdog_exception('islandora_video_streaming_panopto', $ex);
+          return $failureMessage;
+        }
+        $streaming['upload_id'] = $session['ID'];
+      } catch (Exception $ex) {
+        $messages = array(
+          'message' => t('Panopto upload error: @exception'),
+          'message_sub' => array('@exception' => $ex->getMessage()),
+          'type' => 'watchdog',
+          'severity' => WATCHDOG_ERROR,
+        );
+        $failureMessage['messages'][] = $messages;
+        watchdog_exception('islandora_video_streaming_panopto', $ex);
+      }
+    } else {
+      $messages = array(
+        'message' => t('Failed to retrieve session for object: @pid'),
+        'message_sub' => array('@pid' => $object->id),
+        'type' => 'watchdog',
+        'severity' => WATCHDOG_ERROR,
+      );
+      $failureMessage['messages'][] = $messages;
+      return $failureMessage;
+    }
+    file_delete($archival_file['file'], true);
     if ($streaming) {
        // Write the data to a temporary file.
       $temp_name = drupal_tempnam('temporary://', 'file');
       if (file_put_contents($temp_name, json_encode($streaming)) === FALSE) {
-        $failureMessage['messages'][0]['message'] = t('Failed to create temporary file @filename for object @pid when uploading to Panopto.');
-        $failureMessage['messages'][0]['message_sub']['@filename'] = $temp_name;
+        $messages = array(
+            'message' => t("Failed to create temporary file @filename for object @pid when uploading to Panopto"),
+            'message_sub' => array(
+              '@pid' => $object->id, 
+              '@filename' => $temp_name
+            ),
+            'type' => 'watchdog',
+            'severity' => WATCHDOG_ERROR,
+        );
+        $failureMessage['messages'][] = $messages;
         return $failureMessage;
       }
-      return islandora_video_add_datastream($object, 'KALTURA_STREAM', $temp_name);
+      return islandora_video_add_datastream($object, 'PANOPTO_INFO', $temp_name);
     } else {
       return $failureMessage;
     }
   }
 }
 
+/*
+ * Requests token and returns token
+ * @param String $oauthendpoint 
+ */
+function islandora_video_streaming_panopto_get_token($oauthendpoint) {
+  $client_id = variable_get('islandora_video_streaming_panopto_client_id');
+  $secret_client_id = variable_get('islandora_video_streaming_panopto_secret_client_id');
+  $username = variable_get('islandora_video_streaming_panopto_username');
+  $password = variable_get('islandora_video_streaming_panopto_password');
+  
+  $response = null;
+  $token = null;
+  
+  if (isset($client_id) && isset($secret_client_id)) {
```

Comment:
Presumably `$oauthendpoint`, `$username`, and `$password` are also required in addition to the check on `isset($client_id) && isset($secret_client_id)`.

Reply:
Done.